app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TransactionsController < ApplicationController
+  before_action :set_transaction, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /transactions or /transactions.json
+  def index
+    @transactions = Transaction.all
+  end
+
+  # GET /transactions/1 or /transactions/1.json
+  def show; end
+
+  # GET /transactions/new
+  def new
+    @category = Category.find(params[:category_id])
+    @transaction = Transaction.new
+  end
+
+  # GET /transactions/1/edit
+  def edit; end
+
+  # POST /transactions or /transactions.json
+  def create
+    @category = Category.find(params[:category_id])
+    # @transaction = Transaction.new(transaction_params)
+    # @transaction.user_id = current_user.id
```

Comment:
Please remove commented code from here as it can make the whole project unclean.